network/panos/panos_admin.py

```diff
@@ -0,0 +1,204 @@
+    description: success status
+    returned: success
+    type: string
+    sample: "okey dokey"
+'''
+
+try:
+    import pan.xapi
+    HAS_LIB = True
+except ImportError:
+    HAS_LIB = False
+
+_ADMIN_XPATH = "/config/mgt-config/users/entry[@name='%s']"
+
+
+def admin_exists(xapi, admin_username):
+    xapi.get(_ADMIN_XPATH % admin_username)
+    e = xapi.element_root.find('.//entry')
+    return e
+
+
+def admin_set(xapi, module, admin_username, admin_password, role):
+    if admin_password is not None:
+        xapi.op(cmd='request password-hash password "%s"' % admin_password,
+                cmd_xml=True)
+        r = xapi.element_root
+        phash = r.find('.//phash').text
+    if role is not None:
+        rbval = "yes"
+        if role != "superuser" and role != 'superreader':
+            rbval = ""
+
+    ea = admin_exists(xapi, admin_username)
+    if ea is not None:
+        # user exists
+        changed = False
+
+        if role is not None:
+            rb = ea.find('.//role-based')
+            if rb is not None:
+                if rb[0].tag != role:
+                    changed = True
+                    xpath = _ADMIN_XPATH % admin_username
+                    xpath += '/permissions/role-based/%s' % rb[0].tag
+                    xapi.delete(xpath=xpath)
+
+                    xpath = _ADMIN_XPATH % admin_username
+                    xpath += '/permissions/role-based'
+                    xapi.set(xpath=xpath,
+                             element='<%s>%s</%s>' % (role, rbval, role))
+
+        if admin_password is not None:
+            xapi.edit(xpath=_ADMIN_XPATH % admin_username+'/phash',
+                      element='<phash>%s</phash>' % phash)
+            changed = True
+
+        return changed
+
+    # setup the non encrypted part of the monitor
+    exml = []
+
+    exml.append('<phash>%s</phash>' % phash)
+    exml.append('<permissions><role-based><%s>%s</%s>'
+                '</role-based></permissions>' % (role, rbval, role))
+
+    exml = ''.join(exml)
+    # module.fail_json(msg=exml)
+
+    xapi.set(xpath=_ADMIN_XPATH % admin_username, element=exml)
+
+    return True
+
+
+def main():
+    argument_spec = dict(
+        ip_address=dict(default=None),
+        password=dict(default=None, no_log=True),
+        username=dict(default='admin'),
+        admin_username=dict(default='admin'),
+        admin_password=dict(default=None, no_log=True),
+        role=dict(default=None),
+        commit=dict(type='bool', default=True)
+    )
+    module = AnsibleModule(argument_spec=argument_spec)
+
+    if not HAS_LIB:
+        module.fail_json(msg='pan-python required for this module')
+
+    ip_address = module.params["ip_address"]
+    if not ip_address:
+        module.fail_json(msg="ip_address should be specified")
+    password = module.params["password"]
+    if not password:
+        module.fail_json(msg="password is required")
+    username = module.params['username']
+
+    xapi = pan.xapi.PanXapi(
```

Comment:
If you plan to have multiple modules that will implement the same code, consider extracting the common code into a shared module and send a PR to ansible/ansible in the module_utils folder.  Then you will be able to import your shared module instead of having to copy it in each time.